File: OutlierProtege/Models/Source.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OutlierProtege.Models
{
    public class Source
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Description { get; set; }
    }
}

File: OutlierProtege/App/Models/Source.ts
module App.Models {
    "use strict";

    export class Source implements Interfaces.ISource {
        public id: number;
        public description: string;
        constructor(private $q: ng.IQService, private resourceService: Services.IResourceService) {
        }

        public saveToDb(): ng.IPromise<Models.Source> {
            var deferred = this.$q.defer<Models.Source>();
            this.resourceService.source.save(this, (data) => {
                deferred.resolve(<Models.Source>data);
            });
            return deferred.promise;
        }
    }
} 